skbio/io/format/stockholm.py

```diff
@@ -0,0 +1,301 @@
+
+@stockholm.sniffer()
+def _stockholm_sniffer(fh):
+    # Smells a Stockholm file if the first line contains 'Stockholm' and the
+    # version number.
+    line = next(fh)
+    if line[:15] == "# STOCKHOLM 1.0":
+        return True, {}
+
+    return False, {}
+
+
+@stockholm.reader(TabularMSA)
+def _stockholm_to_tabular_msa(fh, constructor=Protein):
+    # Uses OrderedDict() to make sure dna_data isn't arranged randomly
+    dna_data = OrderedDict()
+    metadata = {}
+    positional_metadata = {}
+    seqs = []
+
+    # Retrieves data from file
+    for line in fh:
+        if not (line.startswith("#") or line.startswith("//") or
+                line.isspace() == True):
+            dna_data = _parse_stockholm_line_data(line, dna_data)
+    fh.seek(0)
+
+    # Retrieves metadata from file
+    for line in fh:
+        if line.startswith("#=GF"):
+            metadata = _parse_stockholm_line_gf(line, metadata)
+        if line.startswith("#=GS"):
+            dna_data = _parse_stockholm_line_gs(line, dna_data)
+        if line.startswith("#=GR"):
+            dna_data = _parse_stockholm_line_gr(line, dna_data)
+        if line.startswith('#=GC'):
+            positional_metadata = _parse_stockholm_line_gc(line,
+                                                           positional_metadata)
+
+    for key in dna_data.keys():
+        # Sets blank dictionaries and lists to None instead
+        if not dna_data[key][1]:
+            dna_data[key][1] = None
+        if not dna_data[key][2]:
+            dna_data[key][2] = None
+        # Adds each sequence to the MSA data
+        seqs.append(constructor(dna_data[key][0], metadata=dna_data[key][1],
+                                positional_metadata=dna_data[key][2]))
+
+    if not seqs:
+        raise StockholmFormatError("No data present in file.")
+    if not positional_metadata:
+        positional_metadata = None
+    # Constructs TabularMSA
+    return TabularMSA(seqs, metadata=metadata,
+                      positional_metadata=positional_metadata,
+                      index=list(dna_data.keys()))
+
+
+def _parse_stockholm_line_gf(line, metadata):
+    """Takes ``#=GF`` line and returns parsed data."""
+    line = _remove_newline(line.split(' ', 2))
+    if line[1] in metadata.keys():
+        metadata[line[1]] = metadata[line[1]] + ' ' + line[2]
+    else:
+        metadata[line[1]] = line[2]
+    return metadata
+
+
+def _parse_stockholm_line_gs(line, dna_data):
+    """Takes ``#=GS`` line and returns parsed data."""
+    line = _remove_newline(line.split(' ', 3))
+    if line[1] in dna_data.keys():
+        dna_data[line[1]][1][line[2]] = line[3]
+    else:
+        raise StockholmFormatError("Markup line references nonexistent "
+                                   "data %r." % line[1])
+    return dna_data
+
+
+def _parse_stockholm_line_gr(line, dna_data):
+    """Takes ``#=GR`` line and returns parsed data."""
+    line = _remove_newline(line.split(' '))
+    if len(line) != 4:
```

Comment:
Oops, classic overthinking. The line, after getting parsed kept returning something like ``['#=GC', 'SS_cons', '', '', '', 'CAAGGGAAATACTACGGGAC']`` and for some reason it slipped my mind that I was using ``line.split(' ')`` when I could've been using ``line.split()``. Fixed both here and in GR.